### PR TITLE
Add check for output amount > input amount

### PR DESCRIPTION
**Pull request by eordano**

### CODE REVIEW

**Comment by eordano** [comment]
Closes https://github.com/bitpay/bitcore/issues/1087 and https://github.com/bitpay/bitcore/issues/1049

File: lib/transaction/transaction.js [hunk 1]
```diff
@@ -141,10 +142,32 @@ Transaction.prototype.uncheckedSerialize = Transaction.prototype.toString = func
  *    <li><tt>disableLargeFees</tt>: disable checking for fees that are too large</li>
  *    <li><tt>disableIsFullySigned</tt>: disable checking if all inputs are fully signed</li>
  *    <li><tt>disableDustOutputs</tt>: disable checking if there are no outputs that are dust amounts</li>
+ *    <li><tt>disableMoreOutputThanInput</tt>: disable checking if the transaction spends more bitcoins than the sum of the input amounts</li>
```
Comment (braydonf) [inline lib/transaction/transaction.js]: btw: we don't need html here (markdown will be rendered)

Comment (eordano) [inline lib/transaction/transaction.js]: Thanks for the tip!

File: lib/transaction/transaction.js [hunk 2]
```diff
@@ -136,15 +135,33 @@ Transaction.prototype.uncheckedSerialize = Transaction.prototype.toString = func
...
+ * Retrieve a possible error that could appear when trying to serialize and broadcast this transaction
+ *
+ * @param {Object} opts allows to skip certain tests:
+ * * `disableSmallFees`: disable checking for fees that are too small
+ * * `disableLargeFees`: disable checking for fees that are too large
+ * * `disableIsFullySigned`: disable checking if all inputs are fully signed
+ * * `disableDustOutputs`: disable checking if there are no outputs that are dust amounts
+ * * `disableMoreOutputThanInput`: disable checking if the transaction spends more bitcoins than the sum of the input amounts
+ * @return {bitcore.Error}
+ */
```
Comment (maraoz) [inline lib/transaction/transaction.js]: please avoid repeating the same doc in the 3 methods.

File: test/transaction/transaction.js [hunk 3]
```diff
@@ -388,6 +388,13 @@ describe('Transaction', function() {
             .change(changeAddress);
         }, 'disableIsFullySigned');
       });
+      it('can skip the check that prevents unsigned outputs', function() {
```
Comment (maraoz) [inline test/transaction/transaction.js]: copy-pasta detected :cop:

**Pull request merged**